packages/eslint-mdx/src/helper.ts

```diff
@@ -122,8 +116,8 @@ export function restoreNodeLocation<T>(
   } = node
   const start = range[0] + offset
   const end = range[1] + offset
-  return {
-    ...node,
+
+  return Object.assign(node, {
```

Comment:
is it intentional that the original `node` is modified?

Reply:
Yes, for performance.